mock/pipelines_provider.go

```diff
@@ -0,0 +1,31 @@
+package mock
+
+import (
+	"context"
+
+	fn "knative.dev/kn-plugin-func"
+)
+
+type PipelinesProvider struct {
+	RunInvoked    bool
```

Comment:
I think that the callback are enough and boolean can be left out.
Users of the mock can set such bool values in theirs callbacks.